addons/website_sale/models/website.py

```diff
@@ -1030,6 +1042,29 @@ def _get_canonical_url(self):
     def _get_snippet_defaults(self, snippet):
         return super()._get_snippet_defaults(snippet) | const.SNIPPET_DEFAULTS.get(snippet, {})
 
+    def _prevent_product_sale(self, product_or_template, is_zero_price_product):
+        """Return whether the product should be allowed to sale online or not depending on price
+        and categories.
+
+        :param recordset product_or_template: A record of the product varient or template.
+        :param boolean is_zero_price_product: Whether product price is zero or not.
+        :return: True if the product sale should be prevented, False otherwise
+        :rtype: boolean
+        """
```

Comment:
```suggestion
        """Return whether the product should be allowed to sell online or not depending on price
        and categories.

        :param recordset product_or_template: A record of the product variant or template.
        :param boolean is_zero_price_product: Whether the product price is zero or not.
        :return: True if the product sale should be prevented, False otherwise.
        :rtype: boolean
        """
```
past mistakes 🙈